File: src/Qowaiv.TypeScript/IFormattable.ts
/**
 * Provides functionality to format the value of an object into a string representation.
 */
interface IFormattable
{
    /**
     * Returns a string that represents the object.
     * @returns string.
     */
    toString(): string;

    /**
     * Returns a formatted string that represents the object.
     * @param {string} f The format that this describes the formatting.
     * @returns formatted string.
     */
    format(f: string): string;
}

File: tools/Qowaiv.SvoGenerator/Snippets/IFormattable.cs
#if !NotIFormattable // exec
public partial struct @TSvo : IFormattable
{
    /// <summary>Returns a <see cref="string"/> that represents the @FullName.</summary>
    [Pure]
    public override string ToString() => ToString(format: null, formatProvider: null);

    /// <summary>Returns a formatted <see cref="string"/> that represents the @FullName.</summary>
    /// <param name="format">
    /// The format that describes the formatting.
    /// </param>
    [Pure]
    public string ToString(string? format) => ToString(format, formatProvider: null);

    /// <summary>Returns a formatted <see cref="string"/> that represents the @FullName.</summary>
    /// <param name="provider">
    /// The format provider.
    /// </param>
    [Pure]
    public string ToString(IFormatProvider? provider) => ToString(format: null, provider);
}
#endif // exec
